File: front-end/src/service/NotificationService.ts
import instance from "../util/apiInstance";

export class NotificationService {
    static readPageByAccountId(accountId: string, page = 1, size = 5): Promise<ApiResponse<PageResponse<NotificationResponse>>> {
        return instance.get(`/notification/account/${accountId}`, { params: { page, size } });
    }
}

File: back-end/src/main/java/com/ta2khu75/quiz/service/NotificationService.java
package com.ta2khu75.quiz.service;

import org.springframework.data.domain.Pageable;

import com.ta2khu75.quiz.model.entity.Notification;
import com.ta2khu75.quiz.model.entity.id.NotificationId;
import com.ta2khu75.quiz.model.response.NotificationResponse;
import com.ta2khu75.quiz.model.response.PageResponse;

public interface NotificationService {
	PageResponse<NotificationResponse> readPageByAccountId(String accountId, Pageable pageable);
}
